### PR TITLE
[MagpieTTS] Magpietts longform unify

### PR DESCRIPTION
**Pull request by subhankar-ghosh**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/tts/models/magpietts.py [hunk 1]
```diff
@@ -4782,13 +4789,12 @@ def generate_speech(
 
             predicted_codes = torch.cat(state.all_predictions, dim=-1)  # (B, C, F*T_steps)
             num_steps = len(state.all_predictions)
+            default_frame_len = num_steps * self.frame_stacking_factor
```
Comment (rfejgin) [inline nemo/collections/tts/models/magpietts.py:5179]: Could you add back the comment that was here originally, I think it got lost:
`            # Concatenate the list of predictions along the time dimension. Note that when frame stacking is on, this also undoes the stacking.`

File: nemo/collections/tts/models/magpietts.py [hunk 2]
```diff
@@ -4570,8 +4574,10 @@ def generate_speech(
                 finished_texts_counter={},
                 attn_prior=initial_attn_prior,
             )
+            chunk_end_frame_lens: Dict[int, int] = {}
```
Comment (rfejgin) [inline nemo/collections/tts/models/magpietts.py:4962]: Could you add a comment saying what this tracks and if it persists across chunked calls to generate_speech()? Because it appears that this one keeps state locally, unlike chunk_state which is persistent between calls but not super clear from the naming.

Side note, maybe we could find a better name than `chunk_state` since that structure appears not to be associated with a particular chunk but rather tracks overall inference state (I think). E.g. could call it inference_state or chunk**ed**_inference_state (the latter is admittedly kind of verbose).

File: nemo/collections/tts/models/magpietts.py [hunk 3]
```diff
@@ -4137,9 +4137,10 @@ def _check_eos_and_update_state(
 
         Args:
             chunk_state: Mutable state object tracking history across chunks.
-            audio_codes_next: Sampled audio codes. Shape: (B, num_codebooks).
+            audio_codes_next: Sampled audio codes. Shape: (B, num_codebooks) or (B, num_codebooks, frame_stacking_factor).
```
Comment (rfejgin) [inline nemo/collections/tts/models/magpietts.py]: Isn't it always 3-dimensional, with `frame_stacking_factor` being 1 if there's no frame stacking?

**Review by rfejgin — COMMENTED** [review]
Overall looks good but please see comments.
Also, I earlier confirmed that CERs for the frame stacking model look good after this change (same as pre-unification), so it looks like this fixes the issue we observed.

File: nemo/collections/tts/models/magpietts.py (hunk 2 above)
Comment (subhankar-ghosh) [reply nemo/collections/tts/models/magpietts.py:4962]: Added. It does not maintain state across generate_speech calls. It is local only.

File: nemo/collections/tts/models/magpietts.py (hunk 1 above)
Comment (subhankar-ghosh) [reply nemo/collections/tts/models/magpietts.py:5179]: Added
